src/Components/Shared/src/BaseJSDataStream.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+#if COMPONENTS_SERVER
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+#elif BLAZOR_WEBVIEW
+namespace Microsoft.AspNetCore.Components.WebView
+#elif BLAZOR_WEBASSEMBLY
+namespace Microsoft.AspNetCore.Components.WebAssembly.Infrastructure
+#endif
+{
+    internal abstract class BaseJSDataStream : Stream
```

Comment:
An alternative to deriving would be to give this delegates that this type invokes (my guess is all you need is the Unhandled exception handler). It doesn't look like deriving from this type gives you any particular benefits.

Reply:
Tried this out, however it seems to complicate things quote a bit for WebAssembly/WebView which use a `DotNetObjectReference` of the `Web{Assembly, View}JSDataStream` with platform specific implementations of `ReceiveJSDataChunk`.